server/aesdsocket.c

```diff
@@ -24,532 +24,637 @@
-	if (daemon == 1)
-	{
-		printf("Run as daemon\n\r");
-
-		signal(SIGCHLD, SIG_IGN);
-		signal(SIGHUP, SIG_IGN);
-
-		pid = fork();
-		if (pid == -1)
-			return -1;
-		else if (pid != 0)
-			exit(EXIT_SUCCESS);
-
-		umask(0);
-
-		if (setsid() == -1)
-			return -1;
-
-		if (chdir("/") == -1)
-			return -1;
-
-		close(STDIN_FILENO);
-		close(STDOUT_FILENO);
-		close(STDERR_FILENO);
-
-		//redirect files to /dev/null
-		open("/dev/null", O_RDWR);
-		dup(0);
-		dup(0);
-	}
-
-	//listen for connections
-	rc = listen(sockfd, MAX_CONNECTS);
-	if (rc < 0)
-	{
-		perror("Server listen");
-		goto cleanup;
-	}
-
-	//open /var/tmp/aesdsocketdata file
-	filefd = open(output_file, O_RDWR | O_CREAT, 0644);
-	if (filefd < 0)
-	{
-		perror("Open file");
-		goto cleanup;
-	}
-
-	//for timer functionality
-	if ((daemon == 0) || (pid == 0))
-	{
-		memset(&sev, 0, sizeof(struct sigevent));
-		sev.sigev_notify = SIGEV_THREAD;
-		sev.sigev_notify_function = timer_thread;
-
-		//create a timer
-		if (timer_create(clock_id, &sev, &timerid) != 0)
-		{
-			perror("timer create");
-			goto cleanup;
-		}
-
-		//get start time
-		if (clock_gettime(clock_id, &start_time) != 0)
-		{
-			perror("clock get time");
-			goto cleanup;
-		}
-
-			//define timer interval values
-			struct itimerspec itimerspec;
-			itimerspec.it_interval.tv_sec = 10;
-			itimerspec.it_interval.tv_nsec = 1000000;
-			timespec_add(&itimerspec.it_value, &start_time, &itimerspec.it_interval);
-			if (timer_settime(timerid, TIMER_ABSTIME, &itimerspec, NULL) != 0)
-			{
-				perror("timer set time");
-				goto cleanup;
-			}
-	}
-
-	while (!signal_exit_code)
-	{ //till we get a signal
-
-		addr_size = sizeof(new_addr);
-
-		//accept socket connections
-		new_sockfd = accept(sockfd, (struct sockaddr *)&new_addr, &addr_size);
-		if (new_sockfd < 0)
-		{
-			perror("Server accept");
-			goto cleanup;
-		}
-
-		str = inet_ntoa(new_addr.sin_addr);
-		printf("Accepted connection from %s\n\r", str);
-		syslog(LOG_INFO, "Accepted connection from %s", str);
-
-		//define thread parameters
-		datap = malloc(sizeof(struct slist_data_s));
-		(datap->thread_param).thread_sockfd = new_sockfd;
-		(datap->thread_param).thread_complete = false;
-
-		//insert a link in the list
-		SLIST_INSERT_HEAD(&head, datap, entries);
-
-		//create a thread
-		if (pthread_create(&((datap->thread_param).mythread), NULL, &thread_func, (void *)&(datap->thread_param)) != 0)
-		{
-			printf("error in pthread_create\n");
-			goto cleanup;
-		}
-
-		//search through the linked list to check completed threads
-		SLIST_FOREACH(datap, &head, entries)
-		{
-			if ((datap->thread_param).thread_complete == true)
-			{
-				pthread_join((datap->thread_param).mythread, NULL);
-			}
-		}
-	}
+int port = 9000;
+int rc;
+struct sockaddr_in server_addr, new_addr;
+socklen_t addr_size;
+int daemon = 0;
+pid_t pid;
+int new_sockfd;
+
+struct slist_data_s *datap = NULL;
+
+//remove(output_file);
+
+//check command line arguments
+if (argc == 1)
```

Comment:
@saloni1307 it looks like something happened with your spacing here.  This makes the diff difficult to read.